docs/plugins/README.md

```diff
@@ -0,0 +1,251 @@
+# Go plugin development
+
+For effectively developing go plugins, absorb the following:
+
+- [The official plugin package documentation - Warnings](https://pkg.go.dev/plugin)
+- [Tutorial: Getting started with multi-module workspaces](https://go.dev/doc/tutorial/workspaces)
+
+<TOC>
+
+## Setting up your environment
+
+To develop plugins, you'll need:
+
+- Go based on the gateway version (from go.mod)
+- Git to check out gateway source code
+- A folder with plugins to build
+
+Set up a workspace, which is going to look like:
+
+- `/tyk` - the checkout
+- `/plugins` - the plugins
+
+A sample [Taskfile.yml](Taskfile.yml) is provided here.
+
+```
+task: Available tasks for this project:
+* all:             Do everything
+* checkout:        Checking out tyk
+* workspace:       Create workspace
+* test:            Test plugin build
+* clean:           Clean workspace
+```
+
+Let's break down the individual steps from a run of `task all`.
+
+### 1. Checking out tyk
+
+```
+task: [checkout] git clone --branch release-5.3.6 https://github.com/TykTechnologies/tyk.git tyk-release-5.3.6 || true
+```
+
+The example checkout uses a particular `release-5.3.6` branch, to match a
+release. With newer `git` version, you may pass `--branch v5.3.6` and it
+would use the tag. In case you want to use the tag it's also possible to
+navigate into the folder and issue `git checkout tags/v5.3.6`.
+
+### 2. Preparing the workspace - plugins
+
+```
+task: [workspace] mkdir -p plugins
+task: [workspace:plugins] rm -f go.mod go.sum
+task: [workspace:plugins] go mod init testplugin
+go: creating new go.mod: module testplugin
+task: [workspace:plugins] go mod edit -go "1.22.6"
+task: [workspace:plugins] wget -q https://raw.githubusercontent.com/TykTechnologies/custom-go-plugin/refs/heads/master/go/src/CustomGoPlugin.go
+```
+
+The internal `workspace:plugins` step ensures a few things:
+
+1. create a plugin, create go.mod,
+2. set go.mod go version to what's set in gateway,
+3. have some code to compile in the folder
```

Comment:
I would reorganise a bit these steps:

## Create a plugin:
- create main.go (or what is the name of the file) with your plugin code logic
- create go.mod
- set go.mod to Gateway go.mod version (link to GW go.mod)